src/Compilers/VisualBasic/Test/Semantic/IOperation/IOperationTests.vb

```diff
@@ -352,5 +352,70 @@ BC30581: 'AddressOf' expression cannot be converted to 'Integer' because 'Intege
     IObjectCreationExpression (Constructor: Sub System.Guid..ctor()) (OperationKind.ObjectCreationExpression, Type: System.Guid) (Syntax: 'New System.Guid()')
     IOperation:  (OperationKind.None) (Syntax: 'AddressOf Main')")
         End Sub
+
+        <Fact>
+        <WorkItem(19819, "https://github.com/dotnet/roslyn/issues/19819")>
+        Public Sub UsingDeclarationSyntaxNotNull()
+            Dim source = <compilation>
+                             <file name="c.vb">
+                                 <![CDATA[
+Imports System
+Module Module1
+    Class C1
+        Implements IDisposable
+        Public Sub Dispose() Implements IDisposable.Dispose
+            Throw New NotImplementedException()
+        End Sub
+    End Class
+    Sub S1()
+        Using D1 as New C1()
+        End Using
+    End Sub
+End Module
+]]>
+                             </file>
+                         </compilation>
+
+            Dim comp = CompilationUtils.CreateCompilationWithMscorlibAndVBRuntime(source, parseOptions:=TestOptions.RegularWithIOperationFeature)
+            Dim tree = comp.SyntaxTrees.Single()
+            Dim node = tree.GetRoot().DescendantNodes().OfType(Of UsingBlockSyntax).Single()
+            Dim op = DirectCast(comp.GetSemanticModel(tree).GetOperationInternal(node), IUsingStatement)
+
+            Assert.NotNull(op.Declaration.Syntax)
+            Assert.Equal("Using D1 as New C1()", op.Declaration.Syntax.ToString())
+        End Sub
+
+        <Fact>
+        <WorkItem(19819, "https://github.com/dotnet/roslyn/issues/19819")>
+        Public Sub UsingDeclarationIncompleteUsingNotNullSyntax()
+            Dim source = <compilation>
+                             <file name="c.vb">
+                                 <![CDATA[
+Imports System
+Module Module1
+    Class C1
+        Implements IDisposable
+        Public Sub Dispose() Implements IDisposable.Dispose
+            Throw New NotImplementedException()
+        End Sub
+    End Class
+    Sub S1()
+        Using
+        End Using
+    End Sub
+End Module
+]]>
+                             </file>
+                         </compilation>
+
+            Dim comp = CompilationUtils.CreateCompilationWithMscorlibAndVBRuntime(source, parseOptions:=TestOptions.RegularWithIOperationFeature)
+            Dim tree = comp.SyntaxTrees.Single()
+            Dim node = tree.GetRoot().DescendantNodes().OfType(Of UsingBlockSyntax).Single()
+            Dim op = DirectCast(comp.GetSemanticModel(tree).GetOperationInternal(node), IUsingStatement)
+
+            Assert.NotNull(op.Declaration.Syntax)
+            Assert.Equal("Using", op.Declaration.Syntax.ToString())
```

Comment:
As mentioned above, do we want to return an non-null operation for `Declaration` in this case?